benches/range_string-creation_functions.rs

```diff
@@ -0,0 +1,585 @@
+            chars.into()
+        };
+
+        chars.sort_unstable();
+
+        chars.dedup();
+
+        chars.into_iter().collect()
+        // chars.as_slice().iter().collect()
+    }
+
+    #[allow(unexpected_cfgs)]
+    #[cfg(feature = "auto-buffer")]
+    pub(super) fn range_string_from_slice_6(
+        chars : &[char],
+        flags : i64,
+    ) -> String {
+        let mut chars = if 0 != (shwild::IGNORE_CASE & flags) {
+            let mut ci_chars = auto_buffer::AutoBuffer::<char, 50>::with_capacity(chars.len() * 2);
+
+            for (ix, c) in chars.iter().enumerate() {
+                if c.is_alphabetic() {
+                    ci_chars.push(c.to_ascii_uppercase());
+
+                    ci_chars.push(c.to_ascii_lowercase());
+                } else {
+                    ci_chars.push(*c);
+                }
+            }
+
+            ci_chars
+        } else {
+            chars.into()
+        };
+
+        chars.sort_unstable();
+
+        chars.dedup();
+
+        chars.into_iter().collect()
+    }
+
+    pub(super) fn range_string_from_slice_7(
+        chars : &[char],
+        flags : i64,
+    ) -> String {
+        if 0 != (shwild::IGNORE_CASE & flags) {
+            let mut bytes = Vec::<u8>::with_capacity(chars.len() * 2);
+
+            for c in chars {
+                if c.is_alphabetic() {
+                    bytes.push(c.to_ascii_uppercase() as u8);
+                }
+                bytes.push(c.to_ascii_lowercase() as u8);
+            }
+
+            bytes.sort_unstable();
+
+            bytes.dedup();
+
+            unsafe {
+
+                String::from_utf8_unchecked(bytes)
+            }
+        } else {
+            let mut bytes : Vec<u8> = Vec::with_capacity(chars.len());
+
+            for c in chars {
+                bytes.push(*c as u8);
+            }
+
+            bytes.sort_unstable();
+
+            bytes.dedup();
+
+            unsafe {
+
+                String::from_utf8_unchecked(bytes)
+            }
+        }
+    }
+}
+
+
+mod implementation {
+    use criterion::BenchmarkId;
+
+
+    pub(super) fn make_id(
+        benchmarked_function_name : &'static str,
+        input_string : &str,
+        flags : i64,
+    ) -> BenchmarkId {
+        let parameter =
+        if 0 != (flags & shwild::IGNORE_CASE) {
+            format!("I:'{input_string}'")
+        } else {
+            format!("'{input_string}'")
+        };
+
+        BenchmarkId::new(
+            format!("`{benchmarked_function_name}()`"),
+            parameter,
+        )
+    }
+}
+
+
+
+fn BENCHMARK_range_string_from_slice_X_WITH_(
+    c : &mut Criterion,
+    input_string : &str,
+    flags: i64,
+) {
+    let input = input_string.chars().collect::<Vec<_>>();
+
+    let id = implementation::make_id("range_string_from_slice_X", input_string, flags);
+
+    c.bench_with_input(id, &input, |b, input| {
+        b.iter(|| {
+            let s = utils::range_string_from_slice_X(input.as_slice(), flags);
```

Comment:
I think these inputs here (and everywhere, really) need `black_box` to prohibit any possible inlining.

Reply:
Done